### DISCUSSION TITLE
v0.3.1

### DISCUSSION THREAD

**Discussion by stefansundin** [stefansundin/privatkopiera#57] [Announcements]
Det fanns några små problem i v0.3.0, men det får man räkna med när så mycket ändras. Tack till er som har haft tålamod.

Om du har ett problem med en ny version i framtiden så kan du, om du använder Firefox, installera en gammal version av tillägget [på den här sidan](https://addons.mozilla.org/en-US/firefox/addon/privatkopiera/versions/), så det är ett alternativ om du har problem med uppdateringar i framtiden. Det finns tyvärr inget enkelt sätt att göra detsamma i Chrome.

Ändringar i v0.3.1 ([kodändringar](https://github.com/stefansundin/privatkopiera/compare/v0.3.0...v0.3.1)):

- Bytte filtyp till `.mkv` då det har bättre stöd för undertexter.
- Det åtfrågades att undertexterna borde finnas kvar i menyn, så det är tillbaka. Du kan också ändra filändelsen till `.srt` för att endast ladda ned undertexterna.
- I vissa fall finns nu också ett val i menyn för att endast ladda ned ljud. Du kan också ändra filändelsen till `.mka`, `.m4a`, eller `.aac` för att endast ladda ned ljudet.
- Nu stöds <tt>https://www.svt.se/videoplayer-embed/</tt>.
- Små buggfixar.

Jag har nu lagt in en länk till [FFmpeg's dokumentation](https://ffmpeg.org/ffmpeg.html) på hemsidan. Det finns nog behov att ändra på kommandot i vissa fall, så avancerade användare har nog nytta av detta. Om du har eventuella tips så dela gärna med dig av dem.

Diskutera nedan.

**Comment by JimmyLW** [comment]
Hej, det fungerar inte att ladda ner härifrån 
https://www.svtplay.se/video/29402516/i-hetaste-laget

...sista dagen på SVTPlay... :)

**Reply by stefansundin** [reply]
Hej. Verkar fungera för mig. Får detta kommando:

```
ffmpeg -i "https://svt-vod-secure-2a.akamaized.net/d0/secure/20210101/b7a3d325-8e22-4464-9cd6-59a43a535ad5/hls-video-avc-4/hls-video-avc-4.m3u8?aka_me_session_id=AAAAAAAAAADOMAFgAAAAADr9SsV3Dbb0Nj7aFmxHEINHjxNed1dZb0ReuvE6NhaRiMIHPFvuUAkmEV%2fHCs0+v%2foFeP3OeOzc" -i "https://svt-vod-secure-2a.akamaized.net/d0/secure/20210101/b7a3d325-8e22-4464-9cd6-59a43a535ad5/hls-audio-stereo-1/hls-audio-stereo-1.m3u8?aka_me_session_id=AAAAAAAAAADOMAFgAAAAADr9SsV3Dbb0Nj7aFmxHEINHjxNed1dZb0ReuvE6NhaRiMIHPFvuUAkmEV%2fHCs0+v%2foFeP3OeOzc" -vcodec copy -acodec copy "I hetaste laget.mkv"
```

Dock när jag försöker ladda ned så får jag följande fel från FFmpeg:
```
[aac_adtstoasc @ 0x7ff13b120940] Multiple RDBs per frame with CRC is not implemented. Update your FFmpeg version to the newest one from Git. If the problem still occurs, it means that your file has a feature which has not been implemented.
[matroska @ 0x7ff13a00be00] Error applying bitstream filters to an output packet for stream #1: Not yet implemented in FFmpeg, patches welcome
av_interleaved_write_frame(): Not yet implemented in FFmpeg, patches welcome
```

Så det är inget fel på Privatkopiera. Liknar problemen i #54.

Eftersom felet verkar relaterat till ljudet (`aac_adtstoasc` etc) så testade jag att ta bort `-acodec copy` i hopp om att det skulle gå om ljudströmmer kodas om. I början skrev FFmpeg ut en hel massa fel och till slut ett sista fel angående ljudet (`Invalid data found when processing input`). Sedan så fortsatte FFmpeg att ladda ned videon dock. I slutänden fick jag ned hela videon men bara ungefär 3 minuter ljud.

Kanske är detta något extra skydd som påverkar alla filmer. Både den här och den i #54 är ju amerikanska filmer, så det kanske är något skumt med dem.

**Comment by Bub0** [comment]
Det är surrondljud på denna, kanske behöver säga till ffpmg att ta alla kanaler som .ts eller WAV och sparas som AC3?

Den tors 14 jan. 2021 08:13Stefan Sundin <notifications@github.com> skrev:

**Comment by starwars74** [comment]
Tyvärr, fungerar inte på urplay. 
Annars bra.

**Reply by stefansundin** [reply]
Fixat i v0.3.2.

**Comment by rogerrapp** [comment]
Det genererade ffmpeg kommandot fungerar inte längre.
Om man lägger till option  **_-bsf:a aac_adtstoasc_**  och byter suffix på utfilen från .mkv till .mp4 så fungerar det som tidigare.
Ett alternativ skulle kunna vara att generera två kommandon. Ett för mp4 och ett för mkv.

(Firefox på Ubuntu 16.04 LTS)

**Reply by rogerrapp** [reply]
Från TV4-Play:
Följande fungerar inte:
ffmpeg -i "https://lbs-usp-hls-vod.cmore.se/vod/df81e/jn3w0bg1n11(12534071_ISMUSP).ism/jn3w0bg1n11(12534071_ISMUSP).m3u8?hls_no_audio_only=true" -vcodec copy -acodec copy "16 shots.mkv"

**Reply by rogerrapp** [reply]
Felmeddelande:  Stream #0:0 -> #0:1 (copy)
Could not write header for output file #0 (incorrect codec parameters ?): Invalid argument

Gör man såhär istället:
ffmpeg -i "https://lbs-usp-hls-vod.cmore.se/vod/df81e/jn3w0bg1n11(12534071_ISMUSP).ism/jn3w0bg1n11(12534071_ISMUSP).m3u8?hls_no_audio_only=true" **-bsf:a aac_adtstoasc** -vcodec copy -acodec copy "16 shots.**mp4**"
Fungerar det.

**Reply by stefansundin** [reply]
Du kan faktiskt redan ändra filändelsen till `.mp4` så läggs `-absf aac_adtstoasc` till automatiskt (som i tidigare versioner).

Men du borde kanske fundera på att uppdatera ditt operativsystem, eller installera nyare ffmpeg på något annat sätt.

**Comment by rogerrapp** [comment]
Tack Stefan.
Installerade ffmpeg-4.3.1
Nu fungerar det som det ska.

Jättebra plugin där man slipper TV4 reklam.

Toppklass!

/Roger

Den mån 18 jan. 2021 20:35Stefan Sundin <notifications@github.com> skrev:

**Comment by mippen** [comment]
Vissa fungerar utmärkt men osäker på vad följande fel betyder:

[NULL @ 000001f180a7ea80] Unable to find a suitable output format for 'Helen Sjöholm om kvällens gala- En'
Helen Sjöholm om kvällens gala- En: Invalid argument

ffmpeg -i "https://lbs-usp-hls-vod.cmore.se/vod/27a2e/20g5dxwzcd1(13314191_ISMUSP).ism/20g5dxwzcd1(13314191_ISMUSP).m3u8?hls_no_audio_only=true" -vcodec copy -acodec copy "Helen Sjöholm om kvällens gala- "En euforisk kväll".mkv"

**Reply by mippen** [reply]
Såg nu felet... om ett avsnitt har " med i sitt namn så blir det knas

**Reply by stefansundin** [reply]
Hej. Tror det är den här videon du försöker ladda ned? https://www.tv4play.se/program/efter-fem/helen-sj%C3%B6holm-om-kv%C3%A4llens-gala-en-euforisk-kv%C3%A4ll/13314191

För mig är filnamnet `Helen Sjöholm om kvällens gala- ”En euforisk kväll”.mkv`, dvs det sånna där sneda citationstecken. Försökte du ändra till "vanliga" citationstecken? I det fallet så borde texten i fältet bli röd, något som är tänkt att påpeka att det är tecken där som kanske inte är så bra.

Jag ändrade nästan så att citationstecken ersätts med `\"`, men sedan kom jag ihåg att det tyvärr inte fungerar exakt likadant i Linux och Windows, och just nu går det inte att välja i Privatkopiera vilken kommandotolk man använder.

**Reply by stefansundin** [reply]
Diskussion flyttad till https://github.com/stefansundin/privatkopiera/issues/62
